python/astro_metadata_translator/headers.py

```diff
@@ -100,9 +147,25 @@ def merge_headers(headers, mode="overwrite"):
         # Fill the entries that have multiple differing values
         for key in fill:
             merged[key] = [h[key] if key in h else None
-                           for h in itertools.chain([first], headers)]
+                           for h in itertools.chain([first_hdr], headers)]
```

Comment:
I have clarified the comment above. It doesn't need to be a copy but I want an untouched one. I pull it out of `headers` because that simplifies the logic in the other modes at the expense of making this one look a bit weirder. It can't be `merged` because by this point I've already replaced the value in `merged` in the previous loop.

Reply:
Good point.  Thanks for the clarification.